docs/en/writing-running-appium/image-comparison.md

```diff
@@ -97,6 +99,65 @@ File.open('find_result_visual.png', 'wb') { |f| f<< Base64.decode64(find_result_
 assert File.size? 'find_result_visual.png'
 ```
 
+```javascript
+Typescript / Javascript
+  /*
+     Typescsript code for occurrence comparison using the template matching algorithm.
+     It detects if an image is contained in another image (called the template).
+     The image must have the same scale and look the same.
+
+     official doc:
+     https://github.com/appium/appium/blob/master/docs/en/writing-running-appium/image-comparison.md
+     OpenCV algorithm doc:
+     https://docs.opencv.org/2.4/doc/tutorials/imgproc/histograms/template_matching/template_matching.html
+     official sample code:
+     https://github.com/justadudewhohacks/opencv4nodejs/blob/master/examples/templateMatching.js
+
+     You must install opencv4nodejs using the -g option.
+
+     The Javascript client driver webdriverio does not support (in January 2020) the "-image" strategy implemented in the Appium server. You will have more power and understanding while using openCV directly. Since the appium server is in Javascript, you can do all it does with opencv in your test suite.
+
+     The testing framework mocha can be run with typescript to have async/await.
+     You need to run mocha with those options in the right order and with the associated packages installed:
+     NODE_PATH=/path/to/nodejs TS_NODE_PROJECT=config/tsconfig_test.json --require ts-node/register --require tsconfig-paths/register
+     You will also need to make a basic config/tsconfig_test.json
+  */
+  const cv = require(process.env.NODE_PATH + '/lib/node_modules/opencv4nodejs');
+  const isImagePresent = async () => {
+    /// Take screenshot and read the image
+    const screenImagePath = './appium_screenshot1.png';
+    driver.saveScreenshot(screenImagePath)
+    const screenImage = await cv.imreadAsync(screenImagePath);
+    const likedImagePath = './occurrence1.png';
+
+    // Load images
+    const originalMat = await cv.imreadAsync(screenImagePath);
```

Comment:
this could be done in parallel, like in the appium-support

Reply:
fixed